src/Compilers/Test/Core/Compilation/TestOperationVisitor.cs

```diff
@@ -567,7 +567,8 @@ private void VisitMemberReference(IMemberReferenceOperation operation, IEnumerab
                 if (operation.Member.IsStatic &&
                     operation.Instance is IInstanceReferenceOperation)
                 {
-                    Assert.False(operation.Instance.IsImplicit, $"Implicit {nameof(IInstanceReferenceOperation)} on {operation.Syntax}");
+                    Assert.True(!operation.Instance.IsImplicit || operation.Instance.HasErrors(operation.SemanticModel.Compilation),
```

Comment:
> || operation.Instance.HasErrors(operation.SemanticModel.Compilation) [](start = 62, length = 69)

It feels like this breaks one of the invariants that we have. Why the original condition was fine for a regular member access and a problem for parreterns?

Reply:
It feels to me that we should fix the product instead and maintain the invariant.

---
In reply to: [528948352](https://github.com/dotnet/roslyn/pull/49534#discussion_r528948352) [](ancestors = 528948352)

Reply:
> Why the original condition was fine for a regular member access and a problem for patterns?

Patterns include the implicit instance reference on the object being pattern matched against, even though it's a static member. I can adjust the implementation to instead not include that reference, but it may be useful for things introspecting the operation tree?

---
In reply to: [528952013](https://github.com/dotnet/roslyn/pull/49534#discussion_r528952013) [](ancestors = 528952013,528948352)